cumulus/parachains/runtimes/assets/common/src/matching.rs

```diff
@@ -58,6 +58,24 @@ impl<SelfParaId: Get<ParaId>> ContainsPair<MultiLocation, MultiLocation>
 	}
 }
 
+pub struct FromNetwork<SelfNetworkId>(sp_std::marker::PhantomData<SelfNetworkId>);
+impl<SelfNetworkId: Get<NetworkId>> ContainsPair<MultiLocation, MultiLocation>
+	for FromNetwork<SelfNetworkId>
+{
+	fn contains(&a: &MultiLocation, b: &MultiLocation) -> bool {
+		// `a` needs to be from `b` at least
+		if !a.starts_with(b) {
+			return false
+		}
+
+		match a {
+			MultiLocation { parents: 2, interior } => {
```

Comment:
@claravanstaden 
I am not very sure about `parents: 2` here, better use something like this:

```
                UniversalLocation: Get<InteriorMultiLocation>,
...

                let universal_source = UniversalLocation::get();
		// check remote
		let _ = match ensure_is_remote(universal_source, *a) {
			Ok((network_id, _) => *network_id == ExpectedNetworkId::get(),
```